src/charm.py

```diff
@@ -110,43 +114,47 @@ def _on_webui_pebble_ready(self, event: EventBase) -> None:
             return
         if not self._container.can_connect():
             self.unit.status = WaitingStatus("Waiting for container to be ready")
-            event.defer()
             return
-        if not self._config_file_is_written():
+        if not self._container.exists(path=BASE_CONFIG_PATH):
+            self.unit.status = WaitingStatus("Waiting for storage to be attached")
+            return
+
+        self._write_database_information_in_config_file(event)
+
+        if not self._config_file_exists():
             self.unit.status = WaitingStatus("Waiting for config file to be written")
             return
+
         self._container.add_layer("webui", self._pebble_layer, combine=True)
         self._container.replan()
         self._container.restart(self._service_name)
         self.unit.status = ActiveStatus()
 
-    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
-        """Handle database created event.
+    def _write_database_information_in_config_file(self, event: EventBase) -> None:
```

Comment:
This method does not really need the whole event, let's pass it only what it requires (the url list)

Reply:
and it is possible that the event received is not necessarily the db information.  What happens if the event is a different one?